proto/message_contents/public_key.proto

```diff
@@ -26,6 +27,25 @@ message UnsignedPublicKey {
     }
 }
 
+message SignedPublicKeyV2 {
+    bytes key_bytes = 1;  // embeds an UnsignedPublicKey
+    AssociationProof proof = 2; // signs key_bytes
+}
```

Comment:
> one for new contact bundles (with identity key and pre-key)

Is there a direct need to generate a new ReadKey? or could the existing IdentityKey be used for the time being? -- That is clients search for readKeys in the ContactTopic, and use the existing identityKey. 

It would be great to keep IdentityKey rotation (which is a parallel work stream)  separate if possible.